jetty-http2/http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/HpackContext.java

```diff
@@ -183,26 +183,26 @@
         }
     }
 
-    private int _maxDynamicTableSizeInBytes;
-    private int _dynamicTableSizeInBytes;
     private final DynamicTable _dynamicTable;
     private final Map<HttpField, Entry> _fieldMap = new HashMap<>();
     private final Map<String, Entry> _nameMap = new HashMap<>();
+    private int _maxTableSize;
+    private int _tableSize;
 
-    HpackContext(int maxDynamicTableSize)
+    HpackContext(int maxTableSize)
     {
-        _maxDynamicTableSizeInBytes = maxDynamicTableSize;
-        int guesstimateEntries = 10 + maxDynamicTableSize / (32 + 10 + 10);
+        _maxTableSize = maxTableSize;
+        int guesstimateEntries = 10 + maxTableSize / (32 + 10 + 10);
         _dynamicTable = new DynamicTable(guesstimateEntries);
         if (LOG.isDebugEnabled())
-            LOG.debug(String.format("HdrTbl[%x] created max=%d", hashCode(), maxDynamicTableSize));
+            LOG.debug(String.format("HdrTbl[%x] created max=%d", hashCode(), maxTableSize));
     }
 
     public void resize(int newMaxDynamicTableSize)
     {
         if (LOG.isDebugEnabled())
-            LOG.debug(String.format("HdrTbl[%x] resized max=%d->%d", hashCode(), _maxDynamicTableSizeInBytes, newMaxDynamicTableSize));
-        _maxDynamicTableSizeInBytes = newMaxDynamicTableSize;
+            LOG.debug(String.format("HdrTbl[%x] resized max=%d->%d", hashCode(), _maxTableSize, newMaxDynamicTableSize));
+        _maxTableSize = newMaxDynamicTableSize;
```

Comment:
Isn't this more setting the `capacity` rather than the `maxDynamicTableSize`?
or at least `dynamicTableSize` (without max), the max is set by the SETTINGS frame.

For example the spec says: 

> Protocols that use HPACK determine the maximum size that the encoder
>    is permitted to use for the dynamic table.  In HTTP/2, this value is
>    determined by the SETTINGS_HEADER_TABLE_SIZE setting (see
>    Section 6.5.2 of [HTTP2]).
> 
>    An encoder can choose to use less capacity than this maximum size
>    (see Section 6.3), but the chosen size MUST stay lower than or equal
>    to the maximum set by the protocol.


I would also review the other places where max is used to describe the table size.

Reply:
No, method `resize(int)` explicitly takes the `maxTableSize` as parameter.
We can rename it to `updateMaxTableSize()` or even `setMaxTableSize()`, but it sets the max, not the actual size.
The actual size is only tracked by `add()` and `evict()`.

Reply:
So in QPACK we have
1. the table capacity - how much can be stored in the table
2. the max table capacity - a value set ONLY by the settings frame, which the table capacity cannot be set to be larger than this by the remote side.
3. the table size - the combined size of the entries currently stored in the table

So what I am saying is it seems like 1 & 2 have been merged into the one name of `maxDynamicTableSize` here which is a little confusing.

Reply:
In HPACK we have the same.
If we agree that "max size" == "capacity", then:

* maxTableCapacity is handled by the encoder or the decoder, not by `HpackContext`.
* `HpackContext.maxTableSize == tableCapacity` so `HpackContext` only deals with `tableCapacity` and the actual `tableSize`.

I can rename the fields and the methods if you like, but I thought to leave them unchanged.